### DISCUSSION TITLE
pyo3 fails to build when features = ["extension-module"] is active on intel Macbook

### DISCUSSION THREAD

**Discussion by Haaroon** [PyO3/pyo3#2995] [Questions] [answered]
Hi, I have an intel macbook running Ventura 13.0.1. 

We have an open source project that uses pyo3, raphtory/docbrown. 

If I run `cargo build` with `features = ["extension-module"]` enabled in the `pyraphtory/cargo.toml` for pyo3 it fails to build and throws an architecture error see below. 

If I remove the `features = ["extension-module"]` it builds fine. 

Has anyone else had this issue? 

[log.txt](https://github.com/PyO3/pyo3/files/10860287/log.txt)

**Comment by Haaroon** [comment, marked as answer]
summary of the log 

```
error: linking with `cc` failed: exit status: 1
...
  = note: Undefined symbols for architecture x86_64:
            "_PyBaseObject_Type", referenced from:
                _$LT$pyo3..types..any..PyAny$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::h66172685f2521793 in pyraphtory.23maf5aor9aqejqm.rcgu.o
                _$LT$pyo3..pycell..PyCellBase$LT$U$GT$$u20$as$u20$pyo3..pycell..PyCellLayout$LT$T$GT$$GT$::tp_dealloc::ha558e8bcabd19867 in pyraphtory.2zilj4qpggs72oxu.rcgu.o
                _$LT$pyo3..pyclass_init..PyNativeTypeInitializer$LT$T$GT$$u20$as$u20$pyo3..pyclass_init..PyObjectInit$LT$T$GT$$GT$::into_new_object::inner::h6d6a7fbaba08a8b4 in libpyo3-12173d10ddf6b2e1.rlib(pyo3-12173d10ddf6b2e1.pyo3.744816bb-cgu.15.rcgu.o)
...
            "__Py_TrueStruct", referenced from:
                pyo3_ffi::boolobject::Py_True::h504c0a1b1ccf2173 in pyraphtory.39vcj94grrnlz8ou.rcgu.o
                pyo3_ffi::boolobject::Py_True::ha14a3013ea0960d5 in libpyo3-12173d10ddf6b2e1.rlib(pyo3-12173d10ddf6b2e1.pyo3.744816bb-cgu.3.rcgu.o)
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)

```

**Reply by messense** [reply]
https://github.com/PyO3/pyo3/blob/main/guide/src/faq.md#i-cant-run-cargo-test-or-i-cant-build-in-a-cargo-workspace-im-having-linker-issues-like-symbol-not-found-or-undefined-reference-to-_pyexc_systemerror

(I can't access pyo3.rs at the moment so I'm using the github url.)

**Reply by Haaroon** [reply]
this seemed to have worked for now, but on some macbooks i still need to build with the arg which is annoying, alas
